tuts/pallet-in-own-crate/v2.0.0-alpha.6/index.md

```diff
@@ -0,0 +1,300 @@
+> book](https://doc.rust-lang.org/cargo/guide/creating-a-new-project.html) for an
+> in-depth explanation.
+
+We first add our newly-created crate as a dependency in the node's runtime
+`Cargo.toml`. Then we tell the pallet to only build its `std` feature when the
+runtime itself does, as follows:
+
+**`runtime/Cargo.toml`**
+
+```TOML
+# --snip--
+
+[dependencies.test-pallet]
+default-features = false
+path = '../pallets/test-pallet'
+
+# toward the bottom
+[features]
+default = ['std']
+std = [
+    'test-pallet/std',
+    # --snip--
+]
+```
+
+> You **must** set `default_features = false` so that your runtime will
+successfully compile to Wasm.
+
+Next we will update `runtime/src/lib.rs` to actually use our new runtime
+pallet, by adding a trait implementation with our `test_pallet` and add it in
+our `construct_runtime!` macro.
+
+**`runtime/src/lib.rs`**
+
+```rust
+// add the following code block
+impl test_pallet::Trait for Runtime {
+  type Event = Event;
+}
+
+// --snip--
+construct_runtime!(
+  pub enum Runtime where
+    Block = Block,
+    NodeBlock = opaque::Block,
+    UncheckedExtrinsic = UncheckedExtrinsic
+  {
+    // --snip--
+    // add the following line
+    TestPallet: test_pallet::{Module, Call, Storage, Event<T>},
+  }
+);
+```
+
+## Run Your Node
+
+At this point you have the pallet packaged up as it's own crate and included in
+your node's runtime.
+
+1. Make sure you're back in the node template's root directory, then compile and run your node with:
+
+    ```bash
+    cargo build --release
+    ```
+
+2. Purge any existing dev chain (Enter `y` on prompt):
+
+    ```bash
+    ./target/release/node-template purge-chain --dev
+    ```
+
+3. Start your node:
+
+    ```bash
+    ./target/release/node-template --dev
+    ```
+
+Finally, start the [Polkadot-JS Apps connecting to your local
+node](https://polkadot.js.org/apps/#/explorer?rpc=ws://127.0.0.1:9944) to
+confirm that the pallet is working as expected.
+
+> **Notes:** You can also manually set the node URL in Polkadot-JS Apps by
+navigating to the **Settings** tab, and have the **remote node/endpoint to
+connect to** set to **Local Node**.
+
+## Publish Your Pallet
+
+Once your pallet is no longer in test phase, you should consider publishing
+it to GitHub or crates.io.
+
+### Publishing on GitHub
+To publish on GitHub, you need to [create a GitHub
+repository](https://help.github.com/en/articles/create-a-repo) and [push your
+pallet's code](https://help.github.com/en/articles/pushing-to-a-remote) to it.
+
+### Publishing on Crates.io
+Crates.io allows permissionless publishing. Learn the proceedure following their own guide about [publishing on crates.io](https://doc.rust-lang.org/cargo/reference/publishing.html)
+
+## Updating your Runtime's Dependecy
+With your pallet now published on GitHub, crates.io, or both,
+we can update your runtime to use the code that is published instead of a
+hard-coded file system path.
```

Comment:
I like this section. Given that we're ultimately updating the dependency to grab it from github or crates.io, I'm thinking it makes more sense to clone the pallet template somewhere other than inside the node template.

This doesn't have to be resolved in this PR. But if it isn't we should make a ticket for it.